core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1,3 +1,21 @@
+{{ define "instance" -}}
+{{- $S3URI := self.Parts.s3.Asset.URL -}}
```

Comment:
Especially after seeing your design decision caused the confusing "wrong number of argument" error in an edge-case, I still prefer avoiding maps in exchange for verbosity. 

However, I completely agree that going ahead with my suggestion would do result in more verbose code!
I'll leave the decision to you.